http4k-security/oauth/src/main/kotlin/org/http4k/security/oauth/client/clientFilterExtensions.kt

```diff
@@ -99,14 +104,27 @@ fun ClientFilters.OAuthUserCredentials(clientCredentials: Credentials, userCrede
     }
 }
 
-fun ClientFilters.OAuthClientCredentials(config: OAuthProviderConfig) = OAuthClientCredentials(config.credentials)
+fun ClientFilters.OAuthClientCredentials(
+    config: OAuthProviderConfig,
+    scopes: List<String>
+) = OAuthClientCredentials(config.credentials, scopes)
 
-fun ClientFilters.OAuthClientCredentials(clientCredentials: Credentials) = Filter { next ->
+fun ClientFilters.OAuthClientCredentials(
+    clientCredentials: Credentials,
+    scopes: List<String>
+) = Filter { next ->
     {
         next(
             it.with(
-                requestForm of WebForm()
-                    .with(
+                if (scopes.isNotEmpty())
+                    requestForm of WebForm().with(
+                        grantType of "client_credentials",
+                        clientId of clientCredentials.user,
+                        clientSecret of clientCredentials.password,
+                        scope of scopes.joinToString(separator = " ")
+                    )
+                else
+                    requestForm of WebForm().with(
```

Comment:
Would've loved to no have that duplication, but couldn't come up with another way

Reply:
If scope is an optional lens you can maybe do scope.takeif{it.isNotempty()}.jointostring...